src/Jobs/CreatePostgresUserForTenant.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\DB;
+use Stancl\Tenancy\Database\Contracts\TenantWithDatabase;
+
+class CreatePostgresUserForTenant implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(
+        protected TenantWithDatabase&Model $tenant,
+    ) {
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $centralConnection = config('tenancy.database.central_connection');
+        $centralConnectionPassword = config("database.connections.$centralConnection.password");
+
+        $name = $this->tenant->database()->getUsername() ?? $this->tenant->getTenantKey();
+        $password = $this->tenant->database()->getPassword() ?? $centralConnectionPassword;
+
+        // Create the user only if it doesn't already exist
+        if (! count(DB::select('SELECT usename FROM pg_user WHERE usename = $1', [$name])) > 0) {
+            $formattedStatement = DB::select("SELECT format('CREATE USER %I LOGIN PASSWORD %L', '$name', '$password');")[0]->format;
```

Comment:
Could we use a parameterized query instead of using variables in a string?

Username & password *should* be safe, but there are edge cases where they could come from user input. In which case the user could bypass the quoting I think.

Reply:
I was trying to use parameterized query initially, but that didn't work. I've been looking for a way of using parameterized query here, but I couldn't find anything that worked, and it looks like Postgres doesn't support parameterized placeholders for queries like "CREATE USER x LOGIN PASSWORD y".

I think the current format() approach might be the most we can do, though it's sort of a security risk.

Reply:
Searched a bit and seems like `%I` and `%L` properly escape everything.